http.go

```diff
@@ -204,51 +214,58 @@ func urlToFilename(url string) string {
 	return strings.Join(pathParts, "/")
 }
 
-func (h *RasterHttpServer) processParams(r *http.Request) (*RasterParams, int, error) {
-	var rParams RasterParams
+func (h *RasterHttpServer) processDocumentParams(r *http.Request) (*RasterDocumentParams, int, error) {
+	var docParams RasterDocumentParams
+
+	// Clean up the URL path into a local filename.
+	docParams.Filename = urlToFilename(r.URL.Path)
+	if docParams.Filename == "" || docParams.Filename == "/" {
+		return nil, 404, errors.New("Invalid URL path")
+	}
+
+	docParams.StoragePath = h.cache.GetFileName(docParams.Filename)
+	docParams.Timestamp = timestampForRequest(r)
+
+	return &docParams, 0, nil
+
+}
+
+func (h *RasterHttpServer) processImageParams(r *http.Request) (*RasterImageParams, int, error) {
+	var imgParams RasterImageParams
 	var err error
 
 	// Parse out and handle some HTTP parameters
-	rParams.Page, err = pageForRequest(r)
+	imgParams.Page, err = pageForRequest(r)
 	if err != nil {
 		return nil, 400, err
 	}
 
-	rParams.ImageQuality = imageQualityForRequest(r)
+	imgParams.ImageQuality = imageQualityForRequest(r)
 
-	rParams.Width, err = widthForRequest(r)
+	imgParams.Width, err = widthForRequest(r)
 	if err != nil {
 		return nil, 400, err
 	}
 
-	rParams.ImageType = imageTypeForRequest(r)
+	imgParams.ImageType = imageTypeForRequest(r)
 
-	rParams.Scale, err = scaleForRequest(r)
+	imgParams.Scale, err = scaleForRequest(r)
 	if err != nil {
 		return nil, 400, err
 	}
 
-	// Clean up the URL path into a local filename.
-	rParams.Filename = urlToFilename(r.URL.Path)
-	if rParams.Filename == "" || rParams.Filename == "/" {
-		return nil, 404, errors.New("Invalid URL path")
-	}
-
-	rParams.StoragePath = h.cache.GetFileName(rParams.Filename)
-	rParams.Timestamp = timestampForRequest(r)
-
-	return &rParams, 0, nil
+	return &imgParams, 0, nil
 }
 
-// handleImage is an HTTP handler that responds to requests for pages
-func (h *RasterHttpServer) handleImage(w http.ResponseWriter, r *http.Request) {
+// handleDocument is an HTTP handler that responds to requests for documents
+func (h *RasterHttpServer) handleDocument(w http.ResponseWriter, r *http.Request) {
 
 	// Log some debug timing and send to New Relic
 	defer func(startTime time.Time) {
 		log.Debugf("Total request time: %s", time.Since(startTime))
 	}(time.Now())
 
-	t := h.beginTrace("handleImage")
+	t := h.beginTrace("handleDocument")
```

Comment:
Note: The `handleImage` trace will, from now on, contain just the time it takes to render an image. `handleDocument` will also include the time it takes to fetch the file and load it in memory.

Reply:
@mihaitodor Thanks for clarifications. 

I really like the fact [goconvey](https://github.com/smartystreets/goconvey) is BDD and has the `Convey()` func with explicit messages about what it's supposed to test. Also the `So()` and `Should*` are quite convenient for testing purpose.